CRM/Civicase/Upgrader/Steps/Step0012.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+use CRM_Civicase_Factory_CaseTypeCategoryEventHandler as CaseTypeCategoryEventHandlerFactory;
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
+
+/**
+ * CRM_Civicase_Upgrader_Steps_Step0012 class.
+ */
+class CRM_Civicase_Upgrader_Steps_Step0012 {
+
+  /**
+   * Performs Upgrade.
+   */
+  public function apply() {
+    $this->updateCategoryMenuItems();
+
+    return TRUE;
+  }
+
+  /**
+   * Update Case Type category menu items.
+   */
+  private function updateCategoryMenuItems() {
+    try {
+      // Get all case type categories.
+      // (but not default Cases category and not Awards or Prospects)
+      $params = ['name' => ['NOT IN' => ['Cases', 'Awards', 'Prospecting']]];
```

Comment:
No need to hardcode the exceptions here. Case categories added by extensions are reserved types, so we can just check for where the case type categories are not reserved.